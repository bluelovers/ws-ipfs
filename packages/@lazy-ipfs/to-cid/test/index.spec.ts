import toCID, { toRawCID } from '../index';

test(`toCID:string`, () =>
{
	let cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	let actual = toCID(cid);
	let expected = cid;

	expect(actual.toV0().toString()).toStrictEqual(cid);

	expect(actual).toMatchSnapshot({
		multihash: expect.any(Buffer),
	});

	expect(actual.toV0().toString()).toMatchSnapshot();
	expect(actual.toV1().toString()).toMatchSnapshot();

});

test(`toCID:CID`, () =>
{
	let cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	let expected = toCID(cid);
	let actual = toCID(expected);

	expect(actual.toV0().toString()).toStrictEqual(cid);
	expect(actual).toMatchSnapshot(expected);

	expect(actual.toV0().toString()).toMatchSnapshot();
	expect(actual.toV1().toString()).toMatchSnapshot();

});

test(`toCID:raw`, () =>
{
	let cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	let expected = toCID(cid);
	let actual = toCID({
		"version": 0,
		"codec": "dag-pb",
		"multibaseName": "base58btc",
		multihash: Buffer.from([
			18,  32, 183, 254,   8,  30, 244,  17,
			96, 165, 123,  89,  19,  86,  24,  96,
			118, 229, 238, 199, 116,   2,  56,  83,
			37, 188,  26,   8,  22, 181, 187, 118,
			74, 219
		]),
	});

	expect(actual.toV0().toString()).toStrictEqual(cid);
	expect(actual).toMatchSnapshot(expected);

	expect(actual.toV0().toString()).toMatchSnapshot();
	expect(actual.toV1().toString()).toMatchSnapshot();

});

test(`toCID:raw without multibaseName`, () =>
{
	let cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	let expected = toCID(cid);
	let actual = toCID({
		"version": 0,
		"codec": "dag-pb",
		multihash: Buffer.from([
			18,  32, 183, 254,   8,  30, 244,  17,
			96, 165, 123,  89,  19,  86,  24,  96,
			118, 229, 238, 199, 116,   2,  56,  83,
			37, 188,  26,   8,  22, 181, 187, 118,
			74, 219
		]),
	});

	expect(actual.toV0().toString()).toStrictEqual(cid);
	expect(actual).toMatchSnapshot(expected);

	expect(actual.toV0().toString()).toMatchSnapshot();
	expect(actual.toV1().toString()).toMatchSnapshot();

});
