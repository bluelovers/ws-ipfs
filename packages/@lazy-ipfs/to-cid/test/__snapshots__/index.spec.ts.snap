// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toCID:CID 1`] = `
Object {
  "codec": "dag-pb",
  "multibaseName": "base58btc",
  "multihash": Object {
    "0": 18,
    "1": 32,
    "10": 123,
    "11": 89,
    "12": 19,
    "13": 86,
    "14": 24,
    "15": 96,
    "16": 118,
    "17": 229,
    "18": 238,
    "19": 199,
    "2": 183,
    "20": 116,
    "21": 2,
    "22": 56,
    "23": 83,
    "24": 37,
    "25": 188,
    "26": 26,
    "27": 8,
    "28": 22,
    "29": 181,
    "3": 254,
    "30": 187,
    "31": 118,
    "32": 74,
    "33": 219,
    "4": 8,
    "5": 30,
    "6": 244,
    "7": 17,
    "8": 96,
    "9": 165,
  },
  "version": 0,
}
`;

exports[`toCID:CID 2`] = `"Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"`;

exports[`toCID:CID 3`] = `"zdj7Whp8MXSDAgmwiv8f7UBseBzjYfTXiNm7Nk9m4GpZzoHCW"`;

exports[`toCID:raw 1`] = `
Object {
  "codec": "dag-pb",
  "multibaseName": "base58btc",
  "multihash": Object {
    "0": 18,
    "1": 32,
    "10": 123,
    "11": 89,
    "12": 19,
    "13": 86,
    "14": 24,
    "15": 96,
    "16": 118,
    "17": 229,
    "18": 238,
    "19": 199,
    "2": 183,
    "20": 116,
    "21": 2,
    "22": 56,
    "23": 83,
    "24": 37,
    "25": 188,
    "26": 26,
    "27": 8,
    "28": 22,
    "29": 181,
    "3": 254,
    "30": 187,
    "31": 118,
    "32": 74,
    "33": 219,
    "4": 8,
    "5": 30,
    "6": 244,
    "7": 17,
    "8": 96,
    "9": 165,
  },
  "version": 0,
}
`;

exports[`toCID:raw 2`] = `"Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"`;

exports[`toCID:raw 3`] = `"zdj7Whp8MXSDAgmwiv8f7UBseBzjYfTXiNm7Nk9m4GpZzoHCW"`;

exports[`toCID:raw without multibaseName 1`] = `
Object {
  "codec": "dag-pb",
  "multibaseName": "base58btc",
  "multihash": Object {
    "0": 18,
    "1": 32,
    "10": 123,
    "11": 89,
    "12": 19,
    "13": 86,
    "14": 24,
    "15": 96,
    "16": 118,
    "17": 229,
    "18": 238,
    "19": 199,
    "2": 183,
    "20": 116,
    "21": 2,
    "22": 56,
    "23": 83,
    "24": 37,
    "25": 188,
    "26": 26,
    "27": 8,
    "28": 22,
    "29": 181,
    "3": 254,
    "30": 187,
    "31": 118,
    "32": 74,
    "33": 219,
    "4": 8,
    "5": 30,
    "6": 244,
    "7": 17,
    "8": 96,
    "9": 165,
  },
  "version": 0,
}
`;

exports[`toCID:raw without multibaseName 2`] = `"Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"`;

exports[`toCID:raw without multibaseName 3`] = `"zdj7Whp8MXSDAgmwiv8f7UBseBzjYfTXiNm7Nk9m4GpZzoHCW"`;

exports[`toCID:string 1`] = `
Object {
  "codec": "dag-pb",
  "multibaseName": "base58btc",
  "multihash": Any<Uint8Array>,
  "version": 0,
}
`;

exports[`toCID:string 2`] = `"Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"`;

exports[`toCID:string 3`] = `"zdj7Whp8MXSDAgmwiv8f7UBseBzjYfTXiNm7Nk9m4GpZzoHCW"`;
